README.md

```diff
@@ -1,39 +1,35 @@
-* **[RxJavaTemplate](https://github.com/nimblehq/android-templates/tree/kotlin/RxJavaTemplate)**
+### Prerequisites
 
-## Setup
+- Machine running macOS from [MacStadium](https://www.macstadium.com/) or a physical one
+- Android Studio installation is recommended as it will help to install all the required Java, Android
+  SDK, also the AVD which needed for running UI test
 
-1. Clone or download this repository to your local machine, then extract and open the folder
-2. Run `newproject.sh` script to create a new project with the following inputs:
+### Add custom (self-hosted) runner
 
-```
-  -h, --help                         Display this usage message and exit
-  -t, --template [TEMPLATE]          Select template: "rx" - RxJavaTemplate or "crt" - CoroutineTemplate (i.e. rx)
-  -p, --package-name [PACKAGE_NAME]  New package name (i.e. com.example.package)
-  -n, --app-name [APP_NAME]          New app name (i.e. MyApp, "My App")
-```
+1. Open the **Settings** menu of the repository. Go to **Actions** -> **Runners** -> click on the **Add runner** button
+2. Make sure the Operating System dropdown shows **macOS** as a selected item, then run all of the below commands in your machine in order. Specify configurations when prompted.
+3. (Optional) In the directory where the runner was installed, run `./svc.sh install`. This program is for help checking the status of the runner.
+4. In the directory where the runner was installed, run `./run.sh` to open the runner to listening for the upcoming jobs.
 
-Example:
-- Init a new project with `RxJavaTemplate`
-  `./newproject.sh -t rx -p co.myproject.example -n "My Project"`
+### Giving the ability to run on a self-hosted runner to workflow
 
-- Init a new project with `CoroutineTemplate`
-  `./newproject.sh -t crt -p co.myproject.example -n "My Project"`
+Define `runs-on: self-hosted` to all the jobs we want to be run on a self-hosted runner.
 
-3. Update `android_version_code` and `android_version_name`
-  - `RxJavaTemplate/build.gradle`
-  - `CoroutineTemplate/build.gradle`
+### Known issues
+
+- One runner can only execute one job at a time. To run multiple jobs or workflows simultaneously, we might need to consider adding a new runner in the same or different machine.
+- There is an issue related to `actions/cache` in a low-performance machine as the default `upload-chunk-size` value is too big for such a machine. Consider reducing this value when facing such an issue.
+- If there's an error related to being incapable of finding the Android SDK. We might need to specify the Android SDK path into `ANDROID_SDK_ROOT` environment variable, for example: `export ANDROID_SDK_ROOT=/Users/username/Library/Android/sdk`.
+- In a low-performance machine, Creating an AVD from scratch every time running the Android emulator might be a better solution in terms of execution time as the speed for retrieving cache is low in such a machine.
```

Comment:
```suggestion
- One runner can only execute one job at a time. To run multiple jobs or workflows simultaneously, we need to consider adding a new runner in the same or different machine.
- There is an issue related to `actions/cache` in a low-performance machine since the default `upload-chunk-size` value is too big. So, we can consider reducing this value when facing the slow computing issue.
- In case there's an error related to being incapable of finding the Android SDK. We can resolve it by specifying the Android SDK path into `ANDROID_SDK_ROOT` environment variable, for example: `export ANDROID_SDK_ROOT=/Users/username/Library/Android/sdk`.
- In a low-performance machine, Creating an AVD from scratch every time running the Android emulator might be a better solution in terms of execution time as the speed for retrieving cache is low in such a machine.
```

Reply:
Should we add this into 2nd bullet?
```
- In case we have multiple runner machine, it will run all jobs or workflows parallelly by dispersing the workload to all of the machine. If we want to fix the machine to run especially job, we can add the label into the machine and specify to run on that label instead of self-hosted
```